src/plugins/tracker.ts

```diff
@@ -0,0 +1,85 @@
+import ABTestingService from "@/plugins/ab-testing-service.js"
+
+declare global {
+  interface Window {
+    _paq?: [
+      string | (() => void),
+      (number | string)?,
+      string?,
+      string?,
+      string?
+    ][]
+  }
+}
+
+// * Waits for `window._paq` to become available.
+//   Stops trying after reaching the specified timeout.
+function waitForPaq(timeout = 10000): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const startTime = Date.now()
+
+    const intervalId = setInterval(() => {
+      if (window._paq) {
+        clearInterval(intervalId)
+        resolve()
+      } else if (Date.now() - startTime > timeout) {
+        clearInterval(intervalId)
+        reject(new Error("Timeout reached."))
+      }
+    }, 100)
+  })
+}
+
+function initializeABTestingDimensions() {
+  const ABTestingEnvironment = ABTestingService.getEnvironment()
+  for (const name in ABTestingEnvironment) {
+    const data = ABTestingEnvironment[name]
+    if (data.deleted) {
+      tracker.deleteCustomDimension(data.index)
+    } else {
+      tracker.setCustomDimension(data.index, `${name}/${data.value}`)
+    }
+  }
+}
+
+waitForPaq().then(() => {
+  initializeABTestingDimensions()
+})
```

Comment:
C'est pas bloquant mais j'aurais bien mis ces deux méthodes en async await dans une fonction invoquée immédiatement (ou dans une méthode `initialization()` ) pour pouvoir catch une erreur si y a un soucis d'init

```suggestion
(async () => {
  try {
    await waitForPaq();
    await initializeABTestingDimensions();
  } catch (error) {
    console.error("Error while initializing AB testing:", error);
  }
})();
```

Reply:
Ouais je voulais faire ça, mais comme prettier m'oblige a mettre un `;` avant la toute première parenthèse, j'avais renoncé. T'en pense quoi toi ? 

<img width="609" alt="image" src="https://github.com/betagouv/aides-jeunes/assets/4059803/dab71e29-67b9-46fb-a9c2-12655c007564">

Reply:
Ah oui effectivement ça fait un peu étrange le point virgule au début, mais c'est ok pour moi. Je suis pas opposé à ce que tu conserves ta version non plus